src/sfctl/custom_deployment.py

```diff
@@ -0,0 +1,131 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -----------------------------------------------------------------------------
+
+"""Custom commands for managing Service Fabric Mesh resources"""
+
+from __future__ import print_function
+
+import enum
+import json
+import os
+import shutil
+from knack.util import CLIError
+from sfmergeutility import SFMergeUtility
+
+class ResourceType(enum.Enum):
+    """ Defines the valid yaml resource types
+        which are parseable by CLI
+    """
+    application = 1
+    volume = 2
+    network = 3
+    secret = 4
+    secretValue = 5
+    gateway = 6
+
+def get_resource_type(file_name):
+    """ Gets the resource type from the file name
+    :param str file_name: Path of the file

```

Comment:
Why not call this file_path instead? Should also specify if it is an absolute path, relative path, or both which is allowed as the parameter input

Reply:
Same for function get_resource_name

Reply:
took your feedback from and applied it to the code at it's newest location